plans/upgrade.pp

```diff
@@ -6,12 +6,25 @@
   Optional[String[1]] $master_replica_host = undef,
   Optional[String[1]] $puppetdb_database_replica_host = undef,
 
-  String[1] $version = '2018.1.4',
+  String[1] $version,
+
+  # This parameter exists to enable the use case of running pe_xl::upgrade over
+  # the PCP transport. An orchestrator restart happens during provision
+  # replica. Running `bolt plan run` directly on the master and using local
+  # transport for that node will let the plan to run to completion without
+  # failing due to being disconnected from the orchestrator.
+  Boolean $executing_on_master = false,
```

Comment:
Why do we need to do this though? Wouldn't it be better for the user to just specify the target properly? Instead of not specifying it then telling the plan it's running on the master, who not just have the user supply `local://master.whatever`?

Reply:
It would, yeah. However, at that point we would need to do a little more work to update the plans to use proper Target objects so that we can distinguish between their hostnames / assumed certnames, and the protocol+name used to connect to them. The variables are currently used kind of interchangeably. For example, it wouldn't be good to end up writing a whitelist file that specified "local://master".

I implemented it this way for now to match the pe_xl::configure plan. It should be a straightforward iteration to clean up the code to use `get_targets()` and real target objects, be specific about when we want a name vs. when we want the protocol-inclusive Target object, and remove these `executing_on_master` parameters then.